go/arrow/flight/flightsql/client.go

```diff
@@ -500,6 +522,11 @@ func (c *Client) CancelQuery(ctx context.Context, info *flight.FlightInfo, opts
 		return
 	}
 	defer stream.CloseSend()
+	defer func() {
+		// XXX: assuming server will not return a result and then an error
+		// (or else we need to also try to clean up stmt)
+		err = readUntilEOF(stream)
+	}()
```

Comment:
rather than defer this, why not just do it manually after the next if statement?